sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.institutions.controller.getApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,150 @@
+      //Create new application.
+      const savedApplication = await saveFakeApplication(appDataSource, {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+      });
+
+      const endpoint = `/institutions/application/${savedApplication.id}/student/${student.id}`;
+      const institutionUserToken = await getInstitutionToken(
+        InstitutionTokenTypes.CollegeFUser,
+      );
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(institutionUserToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK)
+        .expect({
+          data: {},
+          id: savedApplication.id,
+          applicationStatus: savedApplication.applicationStatus,
+          applicationNumber: savedApplication.applicationNumber,
+          applicationFormName: "SFAA2022-23",
+          applicationProgramYearID: savedApplication.programYearId,
+        });
+    },
+  );
+
+  it(
+    "Should not have access to get the student application details when student has a submitted application " +
+      "for a different institution (application with location id saved).",
+    async () => {
+      // Arrange
+      // Student has a submitted application to the institution.
+      const student = await saveFakeStudent(appDataSource);
+      //Create new application.
+      const savedApplication = await saveFakeApplication(appDataSource, {
+        institutionLocation: collegeCLocation,
+        student,
+      });
+
+      const endpoint = `/institutions/application/${savedApplication.id}/student/${student.id}`;
+      const institutionUserTokenCUser = await getInstitutionToken(
+        InstitutionTokenTypes.CollegeCUser,
+      );
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(institutionUserTokenCUser, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.FORBIDDEN)
+        .expect({
+          statusCode: 403,
+          message: "Forbidden resource",
+          error: "Forbidden",
+        });
+    },
+  );
+
+  it("Should not get the student application details when application is submitted for different institution ", async () => {
+    // Arrange
+    // Student has a submitted application to the institution.
+    const student = await saveFakeStudent(appDataSource);
+    //Create new application.
+    const savedApplication = await saveFakeApplication(appDataSource, {
+      institutionLocation: collegeCLocation,
+      student,
+    });
+
+    const endpoint = `/institutions/application/${savedApplication.id}/student/${student.id}`;
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.FORBIDDEN)
+      .expect({
+        statusCode: 403,
+        message: "Forbidden resource",
```

Comment:
I made a comment on @dheepak-aot PR that can change this check https://github.com/bcgov/SIMS/pull/1915#discussion_r1183142941